File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include <iostream>
#include <algorithm>
#include <stack>
#define fastio cin.tie(0)->ios_base::sync_with_stdio(0)

using namespace std;

int main(void){
    fastio;
    string s1, s2;
    cin >> s1 >> s2;
    stack<char> st;
    for(int i = 0; i < s1.length(); i++){
        st.push(s1[i]);
        if(s1[i]==s2[s2.length()-1] && st.size()>=s2.length()){
            string tmp = "";
            for(int j = 0; j < s2.length(); j++){
                tmp += st.top();
                st.pop();
            }
            reverse(tmp.begin(),tmp.end());
            if(tmp != s2){
                for(int j = 0; j < tmp.length(); j++)st.push(tmp[j]);
            }
        }
    }
    if(st.empty())cout << "FRULA";
    else{
        string answer = "";
        while(!st.empty()){
            answer += st.top();
            st.pop();
        }
        reverse(answer.begin(),answer.end());
        cout << answer;
    }
}

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.java
import java.util.*;
import java.io.*;

public class Main
{
    static String bomb;
    static char input[];
    static Stack<Character> stack = new Stack<>();
    static StringBuilder sb = new StringBuilder();
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        input = br.readLine().toCharArray();
        bomb = br.readLine();

        int len_b = bomb.length(), len_i = input.length;
        
        for(int j=0;j<len_i;j++) {
            stack.push(input[j]);
            if(stack.size()<len_b || input[j] != bomb.charAt(len_b-1)) continue;
            
            StringBuilder temp = new StringBuilder();
            
            for(int i=0;i<len_b;i++){
                temp.append(stack.pop());
            }
            
            temp.reverse();
            if(!bomb.equals(temp.toString())) {
                for(int i=0;i<temp.length();i++){
                    stack.push(temp.charAt(i));
                }
            }
        }
        
        if(stack.isEmpty()) sb.append("FRULA");
        else{
            Stack<Character> answer = new Stack<>();
            while (!stack.isEmpty()) {
                answer.push(stack.pop());
            }
            while (!answer.isEmpty()) {
                sb.append(answer.pop());
            }
        }
        System.out.print(sb);
    }
}